src/databricks/labs/dqx/profiler/profiler.py

```diff
@@ -674,10 +675,21 @@ def _type_supports_min_max(typ: T.DataType) -> bool:
 
     @staticmethod
     def _round_datetime(value: datetime.datetime, direction: str) -> datetime.datetime:
+        """
+        Rounds a datetime value to midnight based on the specified direction.
+
+        :param value: The datetime value to round.
+        :param direction: The rounding direction ("up" or "down").
+        :return: The rounded datetime value.
+        """
         if direction == "down":
             return value.replace(hour=0, minute=0, second=0, microsecond=0)
         if direction == "up":
-            return value.replace(hour=0, minute=0, second=0, microsecond=0) + datetime.timedelta(days=1)
+            try:
+                return value.replace(hour=0, minute=0, second=0, microsecond=0) + datetime.timedelta(days=1)
+            except OverflowError:
```

Comment:
The change is LGTM but can you please add test for this. Either create new or extend existing, e.g. 
https://github.com/databrickslabs/dqx/blob/main/tests/integration/test_profiler.py#L104